cpp/scripts/compute-sanitizer-to-junit-xml.py

```diff
@@ -0,0 +1,168 @@
+    )
+    args = argparser.parse_args()
+    if args.log and not args.out:
+        args.out = os.path.splitext(args.log)[0] + ".xml"
+    return args
+
+
+preamble = "========= "
+
+
+def cslog_to_junit_converter(cs_logs, outfile, args):
+    testcase_name = ""
+    testcase_error_text = []
+    error_number = 0
+    test_cases = []
+    error_count = 0
+    for src in cs_logs:
+        basename = os.path.splitext(os.path.basename(src))[0]
+        with open(src) as log:
+            for line in log:
+                if line.startswith(preamble):
+                    if line.startswith(preamble + "COMPUTE-SANITIZER"):
+                        pass
+                    elif line.startswith(preamble + "ERROR SUMMARY: "):
+                        error_count = int(line.split(" ")[-2])
+                    elif line.endswith(preamble + "\n"):
+                        classname = (
+                            args.classname
+                            + "."
+                            + basename
+                            + ("." + testcase_name if testcase_name else "")
+                        )
+                        item = ET.Element(
+                            "testcase",
+                            attrib={
+                                "classname": classname,
+                                "name": str(error_number),
+                            },
+                        )
+                        message = testcase_error_text[0][len(preamble):].strip()
+                        failure = ET.SubElement(
+                            item, "failure", attrib={"message": message}
+                        )
+                        data = "".join(testcase_error_text)  # failure.text
+                        failure.append(
+                            ET.Comment(
+                                " --><![CDATA["
+                                + data.replace("]]>", "]]]]><![CDATA[>")
+                                + "]]><!-- "
+                            )
+                        )
+                        test_cases.append(item)
+                    elif line[len(preamble)] != " ":
+                        error_number += 1
+                        testcase_error_text = []
+                        testcase_error_text.append(line)
+                    elif line[len(preamble)] == " ":
+                        testcase_error_text.append(line)
+                    else:
+                        print("Unknown line: " + line)
+                elif line.startswith("[ RUN      ] "):  # gtest testcase
+                    testcase_name = line.split("[ RUN      ] ")[1].strip()
+                elif line.startswith("[       OK ] "):
+                    testcase_name = ""
+                elif line.startswith("[  FAILED  ] "):
+                    testcase_name = ""
+                else:
+                    pass
+                    # raise Exception('unexpected line in compute-sanitizer log: '+line)
```

Comment:
Hmm. If none of the above, should the tool pass? 
I think I can see why we'd leave this comment in. Might be useful to uncomment for debugging.

Reply:
Yes. The log may contain stdout from googletest or pytest which tool will ignore (pytest is not memchecked yet, but will also be too large to add, probably not nightly, may be weekly).
This is added for debug for analyzing only memcheck log. (using `--log-file path`)